sbi/samplers/mcmc/slice_numpy.py

```diff
@@ -197,24 +202,301 @@ def _sample_from_conditional(self, i: int, cxi, rng):
+                    rejected = log_probs[c] < sc["logu"]
+
+                    if rejected:
+                        if sc["xi"] < sc["cxi"]:
+                            sc["lx"] = sc["xi"]
+                        else:
+                            sc["ux"] = sc["xi"]
+                        sc["xi"] = (sc["ux"] - sc["lx"]) * self.rng.rand() + sc["lx"]
+                        sc["next_param"] = np.concatenate(
+                            [
+                                sc["x"][: sc["order"][sc["i"]]],
+                                [sc["xi"]],
+                                sc["x"][sc["order"][sc["i"]] + 1 :],
+                            ]
+                        )
+
+                    else:
+                        if sc["t"] < num_samples:
+                            sc["state"] = "BEGIN"
+
+                            sc["x"] = sc["next_param"].copy()
+
+                            if sc["t"] <= (self.tuning):
+                                i = sc["order"][sc["i"]]
+                                sc["width"][i] += (
+                                    (sc["ux"] - sc["lx"]) - sc["width"][i]
+                                ) / (sc["t"] + 1)
+
+                            if sc["i"] < len(sc["order"]) - 1:
+                                sc["i"] += 1
+
+                            else:
+                                if sc["t"] > self.tuning:
+                                    sc["samples"][sc["t"]] = sc["x"].copy()
+
+                                sc["t"] += 1
+
+                                self.state[c]["i"] = 0
+                                self.state[c]["order"] = list(range(self.n_dims))
+                                self.rng.shuffle(self.state[c]["order"])
+
+                                if self.verbose:
+                                    if sc["t"] % 10 == 0:
+                                        pbar.update(10)  # type: ignore
+
+                        else:
+                            sc["state"] = "DONE"
+
+                if sc["state"] == "DONE":
+                    num_chains_finished += 1
+
+        samples = np.stack([self.state[c]["samples"] for c in range(self.num_chains)])
+
+        return samples
 
 
-if __name__ == "__main__":
-    main()
+def slice_np_parallized(
+    potential_function: Callable,
+    initial_params: torch.Tensor,
+    num_samples: int,
+    thin: int,
+    warmup_steps: int,
+    vectorized: bool,
+    num_workers: int = 1,
+    seed: Optional[int] = None,
+    show_progress_bars: bool = False,
+):
+    num_chains = initial_params.shape[0]
+
+    if seed is not None:
+        torch.manual_seed(seed)
+        seeds = torch.randint(high=2**31, size=(num_chains,))
+    else:
+        seeds = [None] * num_chains
+
+    if not vectorized:  # Sample all chains sequentially
+        all_samples = []
+
+        # Define run function for given input.
+        def run_slice_np(inits, seed):
+            # Seed current job.
+            np.random.seed(seed)
```

Comment:
Consider moving this into the `if seed ... else ...` in L440 above

Reply:
I think it needs to be here such that every workers has its own random state.